flow/connectors/utils/monitoring/monitoring.go

```diff
@@ -306,13 +306,18 @@ func (c *CatalogMirrorMonitor) UpdateStartTimeForPartition(
 	ctx context.Context,
 	runUUID string,
 	partition *protos.QRepPartition,
+	overrideStartTime *time.Time,
```

Comment:
```suggestion
	overrideStartTime time.Time,
```
alternatively, drop the nil check, caller can call `time.Now()`

Reply:
Opted for this